iib/workers/tasks/opm_operations.py

```diff
@@ -190,3 +191,217 @@ def _serve_cmd_at_port(serve_cmd, cwd, port, max_tries, wait_time):
+
+
+def opm_migrate(index_db, base_dir):
+    """
+    Migrate SQLite database to File-Based catalog using opm command.
+
+    :param str index_db: path to SQLite index.db which should migrated to FBC.
+    :param str base_dir: base directory to which the database file should be saved.
+    :return: Returns path to directory containing file-based catalog.
+    :rtype: str
+    """
+    from iib.workers.tasks.utils import run_cmd
+
+    fbc_dir_path = os.path.join(base_dir, 'catalog')
+
+    # It may happen that we need to regenerate file-based catalog
+    # based on updated index.db therefore we have to remove the outdated catalog
+    # to be able to generate new one
+    if os.path.exists(fbc_dir_path):
+        shutil.rmtree(fbc_dir_path)
+
+    cmd = ['opm', 'migrate', index_db, fbc_dir_path]
+
+    run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to migrate index.db to file-based catalog')
+    log.info("Migration to file-based catalog was completed.")
+    return fbc_dir_path
+
+
+def opm_generate_dockerfile(fbc_dir, base_dir, index_db, binary_image=None, dockerfile_name=None):
+    """
+    Generate Dockerfile using opm command and adding index.db to hidden location.
+
+    :param str fbc_dir: directory containing file-based catalog (JSON or YAML files).
+    :param str base_dir: base directory to which the database file should be saved.
+    :param str index_db: path to SQLite index.db which should be put to hidden location in container
+    :param str binary_image: pull spec of binary image in which to build catalog.
+    :param str dockerfile_name: name of generated Dockerfile.
+    :return: Returns path to generated Dockerfile
+    :raises: IIBError when Dockerfile was not generated
+    :rtype: str
+    """
+    from iib.workers.tasks.utils import run_cmd
+
+    df_name = f"{os.path.basename(fbc_dir)}.Dockerfile"
+    if dockerfile_name:
+        dockerfile_path = os.path.join(base_dir, dockerfile_name)
+    else:
+        dockerfile_path = os.path.join(base_dir, df_name)
+
+    if os.path.isfile(dockerfile_path):
+        log.info(
+            "Skipping generation of Dockerfile. "
+            "Dockerfile for file-based catalog already exists at %s",
+            dockerfile_path,
+        )
+        return dockerfile_path
+
+    cmd = [
+        'opm',
+        'alpha',
+        'generate',
+        'dockerfile',
+        os.path.abspath(fbc_dir),
+    ]
+
+    if binary_image:
+        log.info('Using binary image to generate dockerfile %s' % binary_image)
+        cmd.extend(['--binary-image', binary_image])
+
+    run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to generate Dockerfile for file-based catalog')
+
+    df_path = os.path.join(base_dir, df_name)
+    if not os.path.isfile(df_path):
+        error_msg = f"Cannot find generated Dockerfile at {df_path}"
+        log.error(error_msg)
+        raise IIBError(error_msg)
+
+    if dockerfile_name:
```

Comment:
on line no.238 you already use similar if

Reply:
Yes, but at that point the file does not have to exists (is not yet generated by run of opm alpha generate dockerfile)

Reply:
Please let me know if I am wrong but does os.join.path() check if the path exists? If NOT you can just create a string by os.join.path() and reuse it. Later on, check if the file exists by os.path.isfile(). it means that you can reuse `dockerfile_path` variable.

Reply:
It is not checking, just joining two paths. 
I do not want to reuse it because I could not call `os.rename(dockerfile_path_opm_default, dockerfile_path)` on line 274.

Reply:
I see. Thanks